src/ProjectOrigin.WalletSystem.Server/Helpers/GroupClaimsHelper.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using ProjectOrigin.WalletSystem.Server.Models;
+using ProjectOrigin.WalletSystem.Server.Services.REST.v1;
+
+namespace ProjectOrigin.WalletSystem.Server.Helpers;
+
+public static class GroupClaimsHelper
+{
+    public static IEnumerable<AggregationResult> GroupByDay(IEnumerable<ClaimViewModel> claims)
+    {
+        return claims
+            .GroupBy(x => new DateTimeOffset(x.ProductionStart.Date))
+            .Select(x => new AggregationResult
+            {
+                Quantity = x.Sum(y => y.Quantity),
+                Start = x.Key.ToUnixTimeSeconds(),
+                End = x.Key.AddDays(1).ToUnixTimeSeconds()
+            });
+    }
+}
```

Comment:
You have added NotImplementedException-placeholders, in the controller. Why not make the future developers job easier, and make sure, there are placeholders in place, across a vertical slice?

DTO + API Spec
↓
Controller
↓
Helpers/Repositories/Services

![download](https://github.com/project-origin/wallet/assets/55732914/edbcebf9-976e-4a30-8767-a01396f8dd4e)

Reply:
Not sure what you mean.